### PR TITLE
[ci] Add new buildbot with latest driver for Linux test

### PR DESCRIPTION
**Pull request by qiao-bo**
The new buildbot (buildbot10-linux) that has Driver 510 can test Vulkan API version 1.3

### CODE REVIEW

**Comment by qiao-bo** [comment]
@frostming @ailzhang Any idea how to neatly solve this syntax issue? Basically I hope to write
```
    strategy:
      matrix:
        gpu_driver: [driver470, driver510]

    runs-on: [self-hosted, cuda, vulkan, cn, ${{ matrix.gpu_driver }}]
```
to run on two self hosted runners. But Github does not like the syntax and complains (https://github.com/taichi-dev/taichi/actions/runs/2306083544). I know the problem is at the line of `runs-on`. If the matrix var is removed, then it is fine ;/.

Any suggestions would be great!

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -245,9 +245,7 @@ jobs:
     timeout-minutes: 60
     strategy:
       matrix:
-        tags: |
```
Comment (frostming) [inline .github/workflows/testing.yml]: Oh, this `|` turns the following into a multiline string I guess.

**Review by ailzhang — APPROVED** [review]
LGTM!

**Pull request merged**